spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkSQLProperties.java

```diff
@@ -45,6 +45,10 @@ private SparkSQLProperties() {}
       "spark.sql.iceberg.aggregate-push-down.enabled";
   public static final boolean AGGREGATE_PUSH_DOWN_ENABLED_DEFAULT = true;
 
+  // Controls whether to push down limit to Iceberg
+  public static final String LIMIT_PUSH_DOWN_ENABLED = "spark.sql.iceberg.limit-push-down.enabled";
+  public static final boolean LIMIT_PUSH_DOWN_ENABLED_DEFAULT = true;
```

Comment:
[Q] will this config be honoured when we use spark with orc / avro ? I see not should we namespace this to parquet then ?

Reply:
This is not honored for ORC/Avro, but I think we should be able to push the limit down for ORC/Avro. I prefer to keep the current namespace and make it work for ORC/Avro. I will take a look at how ORC/Avro works.